### PR TITLE
Nktchinh

### PR DESCRIPTION
**Pull request by ntkchinh**

### CODE REVIEW

File: assets/js/linear_algebra/dot_product_formula2d.js [hunk 1]
```diff
@@ -112,46 +123,69 @@ function plot(scatter, axis, tt){
...
+           [{text: u.coord.y.toFixed(2), key: 'yu'}]], 
+          [start_coord_x, start_coord_y], 14, 5
+          ),
+      [lines_plus, texts_plus] = lib.text_matrix_to_list(
+          [[{text: '\uFE62', text_color: 0, key: 'fplus'}]],
+          [start_coord_x - 0.3 * w_unit, start_coord_y], 14, 5
+          ),
+      [lines_multi, texts_multi] = lib.text_matrix_to_list(
+          [[{text: '\u00D7', text_color: 0, key: 'fmulti'}],
+           [{text:'\u00D7', text_color: 0, key: 'smulti'}]],
```
Comment (thtrieu) [inline assets/js/linear_algebra/dot_product_formula2d.js]: space

File: assets/js/linear_algebra/dot_product_formula2d.js [hunk 2]
```diff
@@ -112,46 +123,69 @@ function plot(scatter, axis, tt){
...
+                  {x: start_coord_x + 1.55 * w_unit,
+                   y: start_coord_y + 0.45 * h_unit, z: 0,
+                   color: 0}],
+
+      hide_z_texts = [{text: '', x: start_coord_x,
+                       y: start_coord_y + 0.3 * h_unit, key: 'zu', text_opacity: 0},
+                      {text: '', x: start_coord_x + 0.9 * w_unit,
+                       y: start_coord_y + 0.3 * h_unit, key: 'zv', text_opacity: 0}],
+
+      hide_sign_texts = [{text:'\uFE62', x: start_coord_x - 0.3 * w_unit,
```
Comment (thtrieu) [inline assets/js/linear_algebra/dot_product_formula2d.js:159]: space

File: assets/js/linear_algebra/dot_product_formula2d.js [hunk 3]
```diff
@@ -112,46 +123,69 @@ function plot(scatter, axis, tt){
...
+                          y: start_coord_y + 0.3 * h_unit,
+                          text_color: 0, text_opacity: 0, key: 'splus'},
+                         {text:'\u00D7', x: start_coord_x + 0.6 * w_unit,
```
Comment (thtrieu) [inline assets/js/linear_algebra/dot_product_formula2d.js:162]: space

File: assets/js/linear_algebra/lib.js [hunk 4]
```diff
@@ -786,8 +786,8 @@ function text_matrix_to_list(coord_texts, coord,
       matrix_w = (0.02 + 0.60 * numb_of_vector +
                   0.02 * (numb_of_vector - 1) * size/14),
                 // 0.02 from edges * 2; 0.02 between numbers, number: 0.60
```
Comment (thtrieu) [inline assets/js/linear_algebra/lib.js]: rm

File: assets/js/linear_algebra/lib.js [hunk 5]
```diff
@@ -786,8 +786,8 @@ function text_matrix_to_list(coord_texts, coord,
       matrix_w = (0.02 + 0.60 * numb_of_vector +
                   0.02 * (numb_of_vector - 1) * size/14),
                 // 0.02 from edges * 2; 0.02 between numbers, number: 0.60 
-      matrix_h = (0.06 + 0.20 * size_of_space +
-                  0.1 * (size_of_space - 1)) * size/14,
+      matrix_h = (0.06 + 0.2 * size_of_space +
+                  0.2 * (size_of_space - 1)) * size/14,
                 // 0.03 from edges * 2; 0.1 between numbers, number: 0.18
```
Comment (thtrieu) [inline assets/js/linear_algebra/lib.js]: rm

File: assets/js/linear_algebra/dot_product_formula2d.js (hunk 1 above)
Comment (ntkchinh) [reply assets/js/linear_algebra/dot_product_formula2d.js]: ok

File: assets/js/linear_algebra/dot_product_formula2d.js (hunk 2 above)
Comment (ntkchinh) [reply assets/js/linear_algebra/dot_product_formula2d.js:159]: done

File: assets/js/linear_algebra/dot_product_formula2d.js (hunk 3 above)
Comment (ntkchinh) [reply assets/js/linear_algebra/dot_product_formula2d.js:162]: done

File: assets/js/linear_algebra/lib.js (hunk 4 above)
Comment (ntkchinh) [reply assets/js/linear_algebra/lib.js]: done

File: assets/js/linear_algebra/lib.js (hunk 5 above)
Comment (ntkchinh) [reply assets/js/linear_algebra/lib.js]: done

**Comment by thtrieu** [comment]
All clear. thanks!

**Pull request merged**